pkg/rpm/google-cloud-ops-agent.spec

```diff
@@ -38,7 +38,7 @@ The Google Cloud Ops Agent collects metrics and logs from the system.
 %install
 cd %{_sourcedir}
 build_distro=%{dist}
-CODE_VERSION=%{version} BUILD_DISTRO=${build_distro#.} DESTDIR="%{buildroot}" ./build.sh
```

Comment:
Why are you eliminating this logic? This is to make sure that the RPM version gets passed into build.sh; without this it's possible that the RPM version and the version that Ops Agent reports at runtime with will differ.

Reply:
See the commit message of 3328aa6d3adf8e08c33e5c9f1b8675f39bb4a717. Previously, `CODE_VERSION` was always the same as `PKG_VERSION` in our automated builds, so it was a no-op from the automated build standpoint. With this PR, the RPM version is actually a transformed `PKG_VERSION`, which makes the `CODE_VERSION` inconsistent between RPM and DEB builds.

If you want to override the version in manual builds, you could just edit `PKG_VERSION`, right? Can you please give an example of a use case where you'd want the RPM package version (and thus the version reported in the agent metrics) to differ from `PKG_VERSION`?

Reply:
The package version will already be different from the Ops Agent version; distros sometimes add various prefixes when building packages (such as `~focal`). I guess the question is whether we want metrics to report the final package version or the underlying source version. IMO we should report the package version. Also it just feels wrong to have two different layers of the stack independently determine the version number - it's supposed to be the package manager's job to track the version number.

Reply:
The logic I ripped out actually stripped off those extra suffixes. So, regardless of my opinion below, this PR is preserving the original behavior for numeric versions. The only question is whether we should allow overriding the `PKG_VERSION` variable at build time.

I believe that the version compiled into the agent itself really ought to be the agent version, not the package version. Other tools (like [OS inventory manager](https://cloud.google.com/compute/docs/instances/os-inventory-management)) should be available to report the package versions. Looking at it another way, the metrics ought to report the version of the _code_, not the _package_.

Reply:
I guess I'm thinking about cases where the package patches the code. I don't think it's necessarily a valid assumption that the code is the same across all package builds. But I think this is a nit, and we can always change this in the future, so *shrug*.